iterator.go

```diff
@@ -27,7 +27,11 @@ func (i *Iterator) Value() (Event, error) {
 	// apply the event to the aggregate
 	f, found := internal.GlobalRegister.EventRegistered(event)
 	if !found {
-		return Event{}, ErrEventNotRegistered
+		return Event{
+			event:    event,
+			data:     nil,
+			metadata: nil,
+		}, ErrEventNotRegistered
```

Comment:
I wrapping the `ErrEventNotRegistered` error on the wrong level when doing it in the projections, it should have been in the iterator.

```suggestion
		return Event{}, fmt.Errorf("event not registered, aggregate type: %s, reason: %s, global version: %d, %w", event.AggregateType(), event.event.Reason, event.GlobalVersion(), ErrEventNotRegistered)
```